### ISSUE TITLE
Alignment between trk and nifti volume

### ISSUE THREAD

**Issue by arokem** [niivue/niivue#498]
Hello! Thanks for this great project. I am trying to use niivue to create a viewer for a large and openly available dataset of diffusion MRI derivatives that we recently published (https://fcp-indi.s3.amazonaws.com/index.html#data/Projects/HBN/BIDS_curated/derivatives/), see [paper](https://www.nature.com/articles/s41597-022-01695-7) for details. 

For now, I am just sketching something out, hard-coding the URLs to specific files, which I will plan to make something that users can select later on. 

Here is the source, based closely on the niivue trx example: https://github.com/nrdg/hbn_viewer/blob/main/index.html
And here is the web-page: https://nrdg.github.io/hbn_viewer/

As you can see there (I hope), the bundle and the volume data are out of register in the visualization, even though the streamline data is in the same space as the volume. I wonder if niivue requires some additional information about the relationship between the streamlines and the volume that I can provide? Thanks again!

**Comment by arokem** [comment]
OK - a bit more information here. It turns out that both the trk file and the nifti file that I am pointing to have the same affine transform stored with their data. However, if I replace the affine stored in the trk header with the 4x4 identity matrix, they now appear register. I wonder if this is expected?

**Comment by hanayik** [comment]
@arokem, glad you're experimenting with NiiVue! Getting late in my timezone. Just wanted to leave a message saying I can look into this over the next few days. However, @neurolabusc may have some insight as well.

**Comment by neurolabusc** [comment]
@arokem I see you are using NiiVue 0.29.0. Your problem is resolved if you use a recent version of NiiVue. You can try this out with a 0.32.0 live demo and the code below:

```
cd niivue
npm install
 <replace ./niivue/src/index.html with code>
npm run dev
```
LiveView HTML code:
```
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width,initial-scale=1.0">
  <title>NiiVue</title>
  <link rel="stylesheet" href="niivue.css">
</head>
<body>
    <noscript>
      <strong>niivue requires JavaScript.</strong>
    </noscript>
    <header>
      <label for="fiberLengthSlider">Length</label>
      <input type="range" min="1" max="80" value="3" class="slider" id="fiberLengthSlider">
      <label for="fiberDitherSlider">Dither</label>
      <input type="range" min="0" max="10" value="1" class="slider" id="fiberDitherSlider">
      <label for="fiberColor">Choose fiber coloration:</label>
      <select name="fiberColor" id="fiberColor">
        <option value="Global">Global direction</option>
        <option value="Local">Local direction</option>
        <option value="DPV0">First Per Vertex Type (if available)</option>
        <option value="DPS0">First Per Streamline Type (if available)</option>
      </select>
      <label for="fiberDecimation">Fiber reduction:</label>
      <select name="fiberDecimation" id="fiberDecimation">
        <option value="1">100%</option>
        <option value="2">50%</option>
        <option value="4">25%</option>
        <option value="10">10%</option>
      </select>
    </header>
    <main>
      <canvas id="gl1"></canvas>
    </main>
    <footer id="location">
      &nbsp;
    </footer>
    <script type="module" async>
      import { Niivue } from './niivue.js'
      import {NVImage} from './nvimage.js'
      import {NVMesh} from './nvmesh.js'
      var volumeList1 = [{ url: "https://fcp-indi.s3.amazonaws.com/data/Projects/HBN/BIDS_curated/derivatives/afq/sub-NDARAA947ZG5/ses-HBNsiteCBIC/sub-NDARAA947ZG5_ses-HBNsiteCBIC_acq-64dir_space-T1w_desc-preproc_dwi_model-DKI_FA.nii.gz"}]
      function handleLocationChange(data){   
        document.getElementById('location').innerHTML = '&nbsp;&nbsp;'+data.vox[0]+'×'+data.vox[1]+'×'+data.vox[2]
      }
      var slider = document.getElementById("fiberLengthSlider");
      slider.oninput = function() {
        nv1.setMeshProperty(nv1.meshes[0].id, 'fiberLength', this.value)
      }
      var slider2 = document.getElementById("fiberDitherSlider");
      slider2.oninput = function() {
        nv1.setMeshProperty(nv1.meshes[0].id, 'fiberDither', this.value * 0.1)
      }
      var drop = document.getElementById("fiberColor")
      drop.onchange = function() {
        const colorName = document.getElementById("fiberColor").value
        nv1.setMeshProperty(nv1.meshes[0].id, 'fiberColor', colorName)
      }
      var dropD = document.getElementById("fiberDecimation")
      dropD.onchange = function() {
        const stride = document.getElementById("fiberDecimation").value
        nv1.setMeshProperty(nv1.meshes[0].id, 'fiberDecimationStride', stride)
      }
      var nv1 = new Niivue({
        show3Dcrosshair: true,
        onLocationChange:handleLocationChange
        })
      nv1.setRadiologicalConvention(false)
      nv1.attachTo('gl1')
      nv1.setClipPlane([-0.1, 270, 0])
      await nv1.loadVolumes(volumeList1)
      nv1.loadMeshes([{url: "https://fcp-indi.s3.amazonaws.com/data/Projects/HBN/BIDS_curated/derivatives/afq/sub-NDARAA947ZG5/ses-HBNsiteCBIC/clean_bundles/sub-NDARAA947ZG5_ses-HBNsiteCBIC_acq-64dir_space-T1w_desc-preproc_dwi_space-RASMM_model-CSD_desc-prob-afq-AntFrontal_tractography.trk"},])
   </script>
</body>
</html>
```
The screenshot shows NiiVue and TrakVis. Since [TRK was created by TrakVis, it is the reference implementation](https://trackvis.org/docs/?subsect=fileformat). The spatial transform in TRK is unintuitive, and even terrific tools like DSI-studio [have had issues implementing it](https://github.com/tee-ar-ex/trx-python/issues/20). I would encourage you to use a simpler, explicit format like [TRX](https://twitter.com/francoisrheault/status/1531634782915874818?lang=en).

![trk](https://user-images.githubusercontent.com/8930807/215270138-41fbfd4a-1b4a-4adb-a663-b613dd43daaa.jpg)

**Comment by arokem** [comment]
Yes: resolved with version 0.32! Thanks! I wish that we had trx around when this dataset was created... 

BTW: you might want to update the documentation with the new version number. For example in the "usage" section on this page: https://niivue.github.io/niivue/devdocs/index.html.
